demography.md

```diff
@@ -29,34 +29,56 @@ from IPython.display import SVG
 
 ## Population structure
 
+All population structure and demography is passed to {func}`msprime.sim_ancestry` via a {class}`msprime.Demography` object.
+
+```{code-cell} ipython3
+dem = msprime.Demography()
+```
+
 ``msprime`` supports simulation from multiple discrete populations,
-each of which is initialized with a {class}`msprime.Population` object.
+each of which is initialized in {class}`msprime.Demography` via the {meth}`msprime.Demography.add_population` method.
 For each population, you can specify a sample size, an effective population size
 at time = 0, an exponential growth rate and a name.
+In this default setup, the samples will be drawn from time = 0, but see [here](https://tskit.dev/msprime/docs/latest/ancestry.html#sampling-time) if you wish to change this.
 
 *Note: Population structure in ``msprime`` closely follows the model used in the
 ``ms`` simulator.
 Unlike ``ms`` however, all times and rates are specified
 in generations and all populations sizes are absolute (that is, not
-multiples of :math:`N_e`).*
+multiples of ``population_size``).*
 
 Suppose we wanted to simulate three sequences each from two populations
 with a constant effective population size of 500.
 
 ```{code-cell} ipython3
-pop0 = msprime.Population(initial_size=500, growth_rate=0, name='Population0')
-pop1 = msprime.Population(initial_size=500, growth_rate=0, name='Population1')
-samples0 = msprime.SampleSet(num_samples=3, population='Population0')
-samples1 = msprime.SampleSet(num_samples=3, population='Population1')
+dem = msprime.Demography()
+dem.add_population(
+  name="Population0",
+  description="Plotted in red.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem.add_population(
+  name="Population1",
+  description="Plotted in blue.",
+  initial_size=500,
+  growth_rate=0
+  )
+dem  
 ```
-
-You can give these to your {class}`msprime.Demography` object as a list
-using the ``populations`` argument.
-(Note that we no longer need to specify ``Ne`` as we have provided a separate size for each population).
+The {class}`msprime.Demography` object that we've just created can be passed to {func}`msprime.sim_ancestry` via the ``demography`` argument.
+We'll also use the ``samples`` argument to specify how many sample haplotypes we wish to draw from each of these populations.
```

Comment:
haplotypes -> individuals

We're sampling k-ploid individuals now rather than nodes, so haplotypes is misleading I think.